tests/utils.py

```diff
@@ -534,20 +534,27 @@ def makeTempDir(self):
         except FileExistsError:
             pass
 
-    def runAsJava(self, main_code, extra_code=None, args=None):
+    def runAsJava(self, main_code, extra_code=None, args=None, timed=False):
         """Run a block of Python code as a Java program."""
         # Output source code into test directory
         transpiler = Transpiler(verbosity=0)
 
         # Don't redirect stderr; we want to see any errors from the transpiler
         # as top level test failures.
         with capture_output(redirect_stderr=False):
+
+            t1_start = time.perf_counter()
+            t2_start = time.process_time()
+
             transpiler.transpile_string("test.py", main_code)
 
             if extra_code:
                 for name, code in extra_code.items():
                     transpiler.transpile_string("%s.py" % name.replace('.', os.path.sep), adjust(code))
 
+            t1_stop = time.perf_counter()
+            t2_stop = time.process_time()
+
```

Comment:
This won't be benchmarking what you think it is (or, at least, it's not benchmarking the thing that needs to be benchmarked. These timing calls are wrapped around the transpilation process - which is the process of converting Python code to Java bytecode. While this is something that certainly *can* be benchmarked, it's not the thing that is a performance concern - it's *runtime* performance that is a problem. 

So - we need the start and end timers to be executed *in Java*, so that we're performance testing the *execution* of the code, not the *compilation*. 

The simplest version of this would be to wrap the timer calls around the call to subprocess (since that's the part that spawns the actual invocation).

Reply:
Ah-ha.. I see. Thanks! Done